test/jasmine/tests/hover_label_test.js

```diff
@@ -652,3 +652,55 @@ describe('hover after resizing', function() {
         }).then(done);
     });
 });
+
+describe('hover on fill', function() {
+    'use strict';
+
+    afterEach(destroyGraphDiv);
+
+    function assertLabelsCorrect(mousePos, labelPos, labelText) {
+        return new Promise(function(resolve) {
+            mouseEvent('mousemove', mousePos[0], mousePos[1]);
+
+            setTimeout(function() {
+                var hoverText = d3.selectAll('g.hovertext');
+                expect(hoverText.size()).toEqual(1);
+                expect(hoverText.text()).toEqual(labelText);
+
+                var transformParts = hoverText.attr('transform').split('(');
+                expect(transformParts[0]).toEqual('translate');
+                var transformCoords = transformParts[1].split(')')[0].split(',');
+                expect(+transformCoords[0]).toBeCloseTo(labelPos[0], -1, labelText + ':x');
+                expect(+transformCoords[1]).toBeCloseTo(labelPos[1], -1, labelText + ':y');
+
+                resolve();
+            }, constants.HOVERMINTIME);
+        });
+    }
+
+    it('should always show one label in the right place', function(done) {
+        var mock = require('@mocks/scatter_fill_self_next.json');
+        mock.data.forEach(function(trace) { trace.hoveron = 'fills'; });
```

Comment:
It is safer `Lib.extendDeep({}, mock)` (like [here](https://github.com/plotly/plotly.js/blob/f69fc5ff0200132544462e22868e2fa6e7c2b609/test/jasmine/tests/hover_label_test.js#L24)) before using it in case the same mock is use in another suite.